### PR TITLE
Make the templates more readable

### PR DESCRIPTION
**Pull request by leofeyer**
Not sure why some of your `{% if` were `{%- if` or if some of mine should be, too, therefore I created a separate PR.

### CODE REVIEW

File: core-bundle/contao/templates/_new/content_element/list.html.twig [hunk 1]
```diff
@@ -6,5 +6,9 @@
 {% endblock %}
 
 {% block list_item %}
-    {%- if as_editor_view %}{{ item|raw }}{% else %}{{ item|insert_tag|raw }}{% endif -%}
+    {%- if as_editor_view %}
+        {{ item|raw }}
+    {% else %}
+        {{ item|insert_tag|raw }}
+    {% endif -%}
 {% endblock %}
```
Comment (fritzmg) [inline core-bundle/contao/templates/_new/content_element/list.html.twig:14]: I think when we structure it like this, we need

```twig
    {% if as_editor_view %}
        {{- item|raw -}}
    {% else %}
        {{- item|insert_tag|raw -}}
    {% endif %}
```

but I am not sure.

/cc @m-vo @ausi

File: core-bundle/contao/templates/_new/component/_headline.html.twig [hunk 2]
```diff
@@ -21,7 +21,11 @@
 
     <{{ tag_name }}{% block headline_attributes %}{{ headline.attributes|default }}{% endblock %}>
     {%- block headline_inner -%}
-        {% if as_editor_view %}{{ headline.text }}{% else %}{{ headline.text|insert_tag_raw }}{% endif %}
+        {% if as_editor_view %}
+            {{ headline.text }}
+        {% else %}
+            {{ headline.text|insert_tag_raw }}
+        {% endif %}
```
Comment (ausi) [inline core-bundle/contao/templates/_new/component/_headline.html.twig]:
```suggestion
        {%- if as_editor_view -%}
            {{ headline.text }}
        {%- else -%}
            {{ headline.text|insert_tag_raw }}
        {%- endif -%}
```

File: core-bundle/contao/templates/_new/content_element/list.html.twig (hunk 1 above)
Comment (ausi) [reply core-bundle/contao/templates/_new/content_element/list.html.twig:14]: We want to remove all whitespace, but in your version @fritzmg the whitespace before `{% if` and after `endif %}` is not removed. I’d write it like so:

```twig
    {%- if as_editor_view -%}
        {{ item|raw }}
    {%- else -%}
        {{ item|insert_tag|raw }}
    {%- endif -%}
```

File: core-bundle/contao/templates/_new/content_element/table.html.twig [hunk 3]
```diff
@@ -10,5 +10,9 @@
 {% endblock -%}
 
 {%- block table_cell_content %}
-    {%- if as_editor_view %}{{ cell|trim|replace({'\n': '<br>\n'})|raw }}{% else %}{{ cell|trim|replace({'\n': '<br>\n'})|insert_tag|raw }}{% endif -%}
+    {%- if as_editor_view %}
+        {{ cell|trim|replace({'\n': '<br>\n'})|raw }}
+    {% else %}
+        {{ cell|trim|replace({'\n': '<br>\n'})|insert_tag|raw }}
+    {% endif -%}
```
Comment (ausi) [inline core-bundle/contao/templates/_new/content_element/table.html.twig]:
```suggestion
    {%- if as_editor_view -%}
        {{ cell|trim|replace({'\n': '<br>\n'})|raw }}
    {%- else -%}
        {{ cell|trim|replace({'\n': '<br>\n'})|insert_tag|raw }}
    {%- endif -%}
```

**Review by ausi — CHANGES_REQUESTED** [review]

**Comment by ausi** [comment]
Technically it does not matter on which side we add the whitespace control (see https://twig.symfony.com/doc/3.x/templates.html#whitespace-control) so it is up to our coding style I think. 

The following versions are all 100% equivalent:

```twig
{% block foo %}
    {%- if as_editor_view -%}
        {{ headline.text }}
    {%- else -%}
        {{ headline.text|insert_tag_raw }}
    {%- endif -%}
{% endblock %}
```

```twig
{% block foo -%}
    {% if as_editor_view %}
        {{- headline.text -}}
    {% else %}
        {{- headline.text|insert_tag_raw -}}
    {% endif %}
{%- endblock %}
```

```twig
{% block foo -%}
    {% if as_editor_view -%}
        {{ headline.text -}}
    {% else -%}
        {{ headline.text|insert_tag_raw -}}
    {% endif -%}
{% endblock %}
```

```twig
{% block foo %}{% if as_editor_view %}{{ headline.text }}{% else %}{{ headline.text|insert_tag_raw }}{% endif %}{% endblock %}
```

**Comment by fritzmg** [comment]
Imho the first version is the easiest to understand.

File: core-bundle/contao/templates/_new/content_element/list.html.twig [hunk 4]
```diff
@@ -6,5 +6,9 @@
...
 
 {% block list_item %}
-    {%- if as_editor_view %}{{ item|raw }}{% else %}{{ item|insert_tag|raw }}{% endif -%}
+    {%- if as_editor_view %}
+        {{ item|raw }}
+    {% else %}
+        {{ item|insert_tag|raw }}
+    {% endif -%}
```
Comment (ausi) [inline core-bundle/contao/templates/_new/content_element/list.html.twig]:
```suggestion
    {%- if as_editor_view -%}
        {{ item|raw }}
    {%- else -%}
        {{ item|insert_tag|raw }}
    {%- endif -%}
```

**Comment by leofeyer** [comment]
Why do we care about whitespace at all? Except inside a `<pre>` element, it does not really make a difference, does it?

**Comment by m-vo** [comment]
There are some scenarios: For example a list of `inline-block` elements that renders `█ █ █ █` instead of `████` if there are whitespace nodes in between. IIRC that's why the `font-size: 0` "hack" was used (still so in the back end, no?).

Nowadays not much of a problem with Flexbox, though.

**Comment by ausi** [comment]
> Why do we care about whitespace at all? Except inside a `<pre>` element, it does not really make a difference, does it?

There are some differences, e.g. browsers handle `<td></td>` and `<td> </td>` different in some cases, and there is the `:empty` selector that considers whitespace to be non-empty so it makes a difference there too.

IMO for all inline-contexts (like in this PR inside h1, li, td) we should avoid adding extra whitespace in the templates.
I also think it makes the resulting HTML code more readable.
These templates are basic core components that will get reused very often, so I think they should be as robust and clean as possible.

File: core-bundle/contao/templates/_new/content_element/list.html.twig (hunk 1 above)
Comment (leofeyer) [reply core-bundle/contao/templates/_new/content_element/list.html.twig:14]: Changed in 8bd552f5952b80a01c9ef721233a31191f98e8f0.

File: core-bundle/contao/templates/_new/component/_headline.html.twig (hunk 2 above)
Comment (leofeyer) [reply core-bundle/contao/templates/_new/component/_headline.html.twig]: Changed in 8bd552f5952b80a01c9ef721233a31191f98e8f0.

File: core-bundle/contao/templates/_new/content_element/table.html.twig (hunk 3 above)
Comment (leofeyer) [reply core-bundle/contao/templates/_new/content_element/table.html.twig]: Changed in 8bd552f5952b80a01c9ef721233a31191f98e8f0.

File: core-bundle/contao/templates/_new/content_element/list.html.twig (hunk 4 above)
Comment (leofeyer) [reply core-bundle/contao/templates/_new/content_element/list.html.twig]: Changed in 8bd552f5952b80a01c9ef721233a31191f98e8f0.

**Review by ausi — APPROVED** [review]

**Pull request closed without merging**